### PR TITLE
MDS Page Patterns v2

### PR DESCRIPTION
**Pull request by juanca**
https://www.pivotaltracker.com/story/show/177312316

## PR upkeep checklist

- [ ] Change log entry N/A
- [x] Label(s)
- [x] Assignee(s)
- [x] Deployment URL: https://mavenlink.github.io/design-system/page-patterns-v2/
- [x] (When ready for review) Reviewer(s)
- [x] Green Bigmaven CI check: https://github.com/mavenlink/mavenlink/pull/24190
- [ ] DevQA on Bigmaven staging (e.g. does the work need to be imported in the internal design system?) N/A

### CODE REVIEW

File: styleguide/components/components-list/components-list-container.jsx [hunk 1]
```diff
@@ -2,17 +2,8 @@ import PropTypes from 'prop-types';
...
-  if (a.name < b.name) return -1;
-  return 0;
-}
-
 export default function ComponentsListContainer(props) {
-  const shouldAlphabetize = !props.items.some(item => item.heading);
-  const items = shouldAlphabetize ? props.items.sort(sortByName) : props.items;
-
-  return <ComponentsList {...props} items={items} />;
+  return <ComponentsList {...props} items={props.items} />;
```
Comment (Bloodyaugust) [inline styleguide/components/components-list/components-list-container.jsx:6]: Nice removal of unnecessary complexity

File: styleguide/webpack.config.js [hunk 2]
```diff
@@ -15,15 +15,20 @@ module.exports = {
...
+        }],
       },
       {
         test: /\.svg$/,
         include: [
           /src\/svgs\//,
         ],
         use: [
-          { loader: 'svg-sprite-loader' },
+          { loader: 'svg-sprite-loader', options: {} },
```
Comment (Bloodyaugust) [inline styleguide/webpack.config.js:31]: Is this addition necessary?

**Review by Bloodyaugust — APPROVED** [review]
LGTM! Minor questions around some stuff.

File: .eslintrc.js [hunk 3]
```diff
@@ -30,7 +30,7 @@ module.exports = {
     files: [
       '.eslintrc.js',
       '.stylintrc.js',
-      'jest.config.js',
+      '*.config.js',
```
[the remark this answers is not in the document]
Comment (juanca) [reply .eslintrc.js:33]: Any and all configuration. e.g. eslint.config.js styleling.config.js babel.config.js etc. The configuration files are for Node processes so we should lint it as commonjs.

File: styleguide/webpack.config.js (hunk 2 above)
Comment (juanca) [reply styleguide/webpack.config.js:31]: Yeah, `svg-sprite-loader` has a bug with Webpack 5 -- this fix should help any future endeavor at upgrading Webpack. It was identified a while ago and no one has fixed it.

I'm actually now re-considering SVG sprite loading. We don't actually get any real benefit from an SVG sprite sheet in new React code. We might even be able to get better benefits if we take SVG files and dynamically generate JSX for those or something.
